File: app/src/main/java/com/rockon999/android/leanbacklauncher/graphics/ClipCircleDrawable.java
package com.rockon999.android.leanbacklauncher.graphics;

import android.graphics.Canvas;
import android.graphics.ColorFilter;
import android.graphics.Paint;
import android.graphics.drawable.Drawable;

public final class ClipCircleDrawable extends Drawable {
    private final int mColor;

    public ClipCircleDrawable(int color) {
        this.mColor = color;
    }

    public void draw(Canvas canvas) {
        float diameter = (float) Math.max(getBounds().width(), getBounds().height());
        Paint paint = new Paint();
        paint.setColor(this.mColor);
        paint.setAntiAlias(true);
        canvas.drawCircle(diameter / 2.0f, diameter / 2.0f, diameter / 2.0f, paint);
    }

    public void setAlpha(int alpha) {
    }

    public void setColorFilter(ColorFilter cf) {
    }

    public int getOpacity() {
        return -3;
    }
}

File: app/src/main/java/com/amazon/tv/leanbacklauncher/graphics/ClipCircleDrawable.kt
package com.amazon.tv.leanbacklauncher.graphics

import android.graphics.Canvas
import android.graphics.ColorFilter
import android.graphics.Paint
import android.graphics.PixelFormat
import android.graphics.drawable.Drawable
import kotlin.math.max

class ClipCircleDrawable(private val mColor: Int) : Drawable() {
    override fun draw(canvas: Canvas) {
        val diameter = max(bounds.width(), bounds.height())
            .toFloat()
        val paint = Paint()
        paint.color = mColor
        paint.isAntiAlias = true
        canvas.drawCircle(diameter / 2.0f, diameter / 2.0f, diameter / 2.0f, paint)
    }

    override fun setAlpha(alpha: Int) {}
    override fun setColorFilter(cf: ColorFilter?) {}
    override fun getOpacity(): Int {
        return PixelFormat.TRANSLUCENT
    }
}